packages/next/src/client/components/react-dev-overlay/internal/components/dev-overlay-no-throw-error-boundary.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react'
+
+export class DevOverlayNoThrowErrorBoundary extends React.Component<{
+  children: React.ReactNode
+}> {
+  state = { error: null }
+
+  static getDerivedStateFromError(error: Error) {
+    return { error }
+  }
+
+  componentDidCatch(error: Error) {
+    // TODO: figure out how to better handle errors thrown inside the dev overlay itself
+    // this is a temporary solution to prevent the dev overlay from causing an
+    // infinite loop of errors
```

Comment:
I think that's actually fine. The last resort error boundary needs to be minimal because the more complex, the more likely it'll also throw.

It's one of those instances where we'd inject additional telemetry for internal usage.